espnet/nets/pytorch_backend/transformer/subsampling_without_posenc.py

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+# Copyright 2020 Emiru Tsunoo
+#  Apache 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
+
+"""Subsampling layer definition."""
```

Comment:
Don't repeat similar classes. These can be generalized like
```python
class Conv2dSubsampling(torch.nn.Module):

    def __init__(self, idim, odim, dropout_rate, kernels, strides):
        assert len(kernels) == len(strides)
        super().__init__()
        olen = idim
        conv = []
        for i, (k, s) in enumerate(zip(kernels, strides)):
            conv += [torch.nn.Conv2d(1 if i == 0 else odim, odim, k, s), torch.nn.ReLU()]
            # https://pytorch.org/docs/stable/generated/torch.nn.Conv2d.html
            olen = math.floor((olen - k) / s + 1)
        self.conv = torch.nn.Sequential(*conv)
        self.out = torch.nn.Linear(odim * olen, odim)
        self.strides = strides
        self.kernels= kernels

    def forward(self, x, x_mask):
        x = x.unsqueeze(1)  # (b, c, t, f)
        x = self.conv(x)
        b, c, t, f = x.size()
        x = self.out(x.transpose(1, 2).contiguous().view(b, t, c * f))
        if x_mask is None:
            return x, None
        for k, s in zip(self.kernels, self.strides):
            x_mask = x_mask[:, :, :-k+1:s]
        return x, x_mask

# Conv2dSubsamplingWOPosEnc is equivalent to
Conv2dSubsampling(..., kernels=[3, 3], strides=[2, 2])
# Conv2dSubsampling6WOPosEnc is equivalent to
Conv2dSubsampling(..., kernels=[3, 5], strides=[2, 3])
# Conv2dSubsampling8WOPosEnc is equivalent to
Conv2dSubsampling(..., kernels=[3, 3, 3], strides=[2, 2, 2])
```

Reply:
Brilliant idea!  I will fix it.